tests/test_utils.py

```diff
@@ -61,3 +61,35 @@ def test_skips_nones(self):
         }
 
         self.assertDictEqual(filter_none(d), {'a': 1})
+
+
+class TestSQLQuery(TestCase):
+    def test_marks_simple_queries_as_safe(self):
+        query = SQLQuery("SELECT * FROM users WHERE userid='{{userid}}'")
+        query.apply({"userid": 22})
+
+        self.assertTrue(query.is_safe())
+
+    def test_marks_multiple_simple_queries_as_safe(self):
+        query = SQLQuery("SELECT * FROM users WHERE userid='{{userid}}' ; SELECT * FROM profiles")
+        query.apply({"userid": 22})
+
+        self.assertTrue(query.is_safe())
+
+    def test_marks_tautologies_as_not_safe(self):
+        query = SQLQuery("SELECT * FROM users WHERE userid={{userid}}")
+        query.apply({"userid": "22 OR 1==1"})
+
+        self.assertFalse(query.is_safe())
+
+    def test_marks_union_queries_as_not_safe(self):
+        query = SQLQuery("SELECT * FROM users WHERE userid={{userid}}")
+        query.apply({"userid": "22 UNION SELECT body, results, 1 FROM reports"})
+
+        self.assertFalse(query.is_safe())
+
+    def test_marks_comment_attacks_as_not_safe(self):
+        query = SQLQuery("SELECT * FROM users WHERE username='{{username}}' AND password='{{password}}'")
+        query.apply({"username": "admin' --"})
+
+        self.assertFalse(query.is_safe())
```

Comment:
Additional cases to consider:

1. Use parameter for column name or order clause.
2. Adding a whole different query. Like the union test just without UNION.

Reply:
Aren't there Data Sources that aren't using SQL? Might sqlparse not fail for those?